compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/JvmLower.kt

```diff
@@ -107,6 +107,7 @@ val jvmPhases = namedIrFilePhase(
             jvmTypeOperatorLoweringPhase then
             foldConstantLoweringPhase then
             flattenStringConcatenationPhase then
+            computeStringTrimPhase then
```

Comment:
I wanted to add this for JS but I don't know how to test it. Would appreciate a pointer so I can follow-up.

And for native I assume I wait until they update to a version of compiler that contains this phrase and then I can update their list?

Reply:
@romanart @bashor ^^

Reply:
K/JS pipeline is defined in [JsLoweringPhases.kt:349](https://github.com/JetBrains/kotlin/blob/a16ca5e66cd3c17a43d15bd4112da86f1b43059d/compiler/ir/backend.js/src/org/jetbrains/kotlin/ir/backend/js/JsLoweringPhases.kt#L349)
You can run JS test using following command `./gradlew :js:js.tests:jsIrTest`.

For JS IR BE we don't have yet infrastructure for such tests. For the current BE we have special directives for that, like `CHECK_NOT_CALLED` (e.g. [inlineCallNoInline.kt:4](https://github.com/JetBrains/kotlin/blob/b389e29db567024cb90e7b1f395e8f47e59014af/js/js.translator/testData/box/inline/inlineCallNoInline.kt#L4)) and `CHECK_NOT_CALLED_IN_SCOPE`(e.g. [lambdaInLambda.kt:5](https://github.com/JetBrains/kotlin/blob/b389e29db567024cb90e7b1f395e8f47e59014af/js/js.translator/testData/box/inline/lambdaInLambda.kt#L5)) and we going to support/adopt them for JS IR BE. You can try to do something in that direction but I think you will face issues in other places unrelated to this PR.

Reply:
Alternatively, you can write something like that:

```
val f = {
    """
        Hello,
        World
    """.trimIndent()
}

fun box(): String {
    if (f.toString().contains("(")) return "fail"
    return "OK"

}
```

Reply:
if you use the last way please leave a TODO that it should be rewritten it using `CHECK_NOT_CALLED`

Reply:
Thanks for the info! If you don't mind, I'll investigate this as a follow-up change.